services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -131,8 +156,18 @@ getUserStatus (_zusr ::: tid ::: uid ::: _) = do
       pure $ UserLegalHoldStatusResponse status (Just lastKey) (Just clientId)
 
 -- | Request to provision a device on the legal hold service for a user
-requestDevice :: UserId ::: TeamId ::: UserId ::: JSON -> Galley Response
-requestDevice (zusr ::: tid ::: uid ::: _) = do
+requestDeviceH :: UserId ::: TeamId ::: UserId ::: JSON -> Galley Response
+requestDeviceH (zusr ::: tid ::: uid ::: _) = do
+  requestDevice zusr tid uid <&> \case
+    RequestDeviceSuccess -> empty & setStatus status201
+    RequestDeviceAlreadyPending -> empty & setStatus status204
+
+data RequestDeviceResult
```

Comment:
If it's a response I think it should be in `galley-types` no?

Reply:
I also wonder if we can't just return `userLHStatus` below, and not introduce a new type in the first place?  That would shift exception handling around, not sure what I think of that either.

I say if in doubt, don't change this at all!  :)

Reply:
Makes sense, there also some more types like this here and in brig.

Ideally, there would be some kind of `galley-api` package where we put those types and routes and also the mapping to status codes you see above. `galley-types` is kind of like that, but it doesn't seem like the status code mapping fits there are the moment.

Reply:
So should we leave it for now?